src/time/time.c

```diff
@@ -0,0 +1,18 @@
+#include <time.h>
+//#include "syscall.h"
+
+__attribute__((eosio_wasm_import))
+extern uint64_t current_time();
+
+time_t time(time_t *t)
+{
+	/*
+		struct timespec ts;
+	__clock_gettime(CLOCK_REALTIME, &ts);
+	if (t) *t = ts.tv_sec;
+	return ts.tv_sec;
+	*/
+	uint32_t elapsed = (uint32_t)(current_time() / 1000000ll);
```

Comment:
Is there a reason for the cast to uint32_t?  time_t is declared as long.

Reply:
I thought this is a substitute of eosio::time_point_sec and its underlying field is uint32_t. I agree with you that there's no reason to cast here.